src/lib/y2network/connection_config/base.rb

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/interface"
+
+module Y2Network
+  module ConnectionConfig
+    # This class is reponsible of a connection configuration
+    class Base
+      # @return [Interface, nil]
+      attr_accessor :interface
+      # @return [String] Bootproto (static, dhcp, none)
+      attr_accessor :bootproto
```

Comment:
I play now with bootproto and I think that ideally this should not be just writter, but be a bit smarter. E.g. for dhcp explicitelly clear ip address and netmask nor hostname and so on.

Reply:
We could think on it when adding the new dialogs / workflow. Currently its just associated with bootproto ifcfg attribute but we could add another abstraction also the configuration associated with dhcp itself...